### PR TITLE
[processor/remoteobserver] rename to remotetapprocessor

### LINKED ISSUE #27873: Rename remoteobserverprocessor to remotetapprocessor
### Component(s)

processor/remoteobserver

### Describe the issue you're reporting

Rename remoteobserverprocessor to remotetapprocessor - observer is used as a set of processors to observe and dynamically configure the collector, so this naming collides.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Rename remoteobserverprocessor to remotetapprocessor

**Link to tracking Issue:**
Fixes #27873

### CODE REVIEW

**Review by MovieStoreGuy — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]
I should know this, but will the label for GitHub issues/PRs now be `processor/remotetap` as a result of this change?

**Review by songy23 — COMMENTED** [review]
>     --- FAIL: TestDefaultProcessors/remoteobserver (0.00s)
> panic: runtime error: invalid memory address or nil pointer dereference [recovered]

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6584479405/job/17889170691?pr=27874#step:9:222 please fix

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -258,7 +258,7 @@ receiver/udplogreceiver/                                                @open-te
 receiver/vcenterreceiver/                                               @open-telemetry/collector-contrib-approvers @djaglowski @schmikei
 receiver/wavefrontreceiver/                                             @open-telemetry/collector-contrib-approvers @samiura
 receiver/webhookeventreceiver/                                          @open-telemetry/collector-contrib-approvers @atoulme @shalper2
-receiver/windowseventlogreceiver/                                       @open-telemetry/collector-contrib-approvers @djaglowski @armstrmi @pjanotti
+receiver/windowseventlogreceiver/                                       @open-telemetry/collector-contrib-approvers @djaglowski @armstrmi
```
[the remark this answers is not in the document]
Comment (atoulme) [reply .github/CODEOWNERS:261]: the codeowners file is out of alignment related to what metadata.yaml says

[the remark this answers is not in the document]
Comment (crobert-1) [reply .github/CODEOWNERS:261]: I see what happened, the `metadata.yaml` wasn't updated with the [change adding this code owner](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27667), so it's out of sync. I'll open a chore PR and this error should go away, and keep the code owner.

[the remark this answers is not in the document]
Comment (crobert-1) [reply .github/CODEOWNERS:261]: #27887 will resolve this issue, and then we can keep @pjanotti as code owner here.

File: .chloggen/rename_remoteobserverprocessor.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (codeboten) [inline .chloggen/rename_remoteobserverprocessor.yaml:4]: is it a breaking change if the module was never tagged?

**Review by codeboten — COMMENTED** [review]
Looks ok, just one question

File: .chloggen/rename_remoteobserverprocessor.yaml (hunk 2 above)
Comment (atoulme) [reply .chloggen/rename_remoteobserverprocessor.yaml:4]: It was tagged a couple times

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -61,7 +61,7 @@ exporter/fileexporter/                                                  @open-te
 exporter/googlecloudexporter/                                           @open-telemetry/collector-contrib-approvers @aabmass @dashpole @jsuereth @punya @damemi @psx95
 exporter/googlecloudpubsubexporter/                                     @open-telemetry/collector-contrib-approvers @alexvanboxel
 exporter/googlemanagedprometheusexporter/                               @open-telemetry/collector-contrib-approvers @aabmass @dashpole @jsuereth @punya @damemi @psx95
-exporter/honeycombmarkerexporter/                                             @open-telemetry/collector-contrib-approvers @fchikwekwe @TylerHelmuth
+exporter/honeycombmarkerexporter/                                       @open-telemetry/collector-contrib-approvers @TylerHelmuth @fchikwekwe
```
Comment (MovieStoreGuy) [inline .github/CODEOWNERS:64]: Linter fix?

File: .github/CODEOWNERS (hunk 1 above)
[the remark this answers is not in the document]
Comment (MovieStoreGuy) [reply .github/CODEOWNERS:261]: I suggest fixing this outside of the PR to help keep the history clean

**Review by MovieStoreGuy — COMMENTED** [review]
Just a couple updates from me.

File: .github/CODEOWNERS (hunk 3 above)
Comment (atoulme) [reply .github/CODEOWNERS:64]: yep.

**Comment by atoulme** [comment]
> I should know this, but will the label for GitHub issues/PRs now be processor/remotetap as a result of this change?

Yes, should be renamed.

**Pull request merged**